controllers/front/OrderConfirmationController.php

```diff
@@ -81,11 +81,14 @@ public function init()
     public function initContent()
     {
         parent::initContent();
+        $order = new Order(Order::getOrderByCartId((int)($this->id_cart)));
+        $order->url_to_invoice = HistoryController::getUrlToInvoice($order, $this->context);
 
         $this->context->smarty->assign(array(
             'is_guest' => $this->context->customer->is_guest,
             'HOOK_ORDER_CONFIRMATION' => $this->displayOrderConfirmation(),
-            'HOOK_PAYMENT_RETURN' => $this->displayPaymentReturn()
+            'HOOK_PAYMENT_RETURN' => $this->displayPaymentReturn(),
+            'order' => $order
```

Comment:
Please don't pass `ObjectModel`'s to templates, as really dangerous actions could be performed on such an object, especially an order.

Please either serialize the whole thing as array (see `Adapter\ObjectSerializer`) or choose the properties you need in the template yourself.